common/src/main/java/org/cloudfoundry/identity/uaa/authorization/UaaAuthorizationEndpoint.java

```diff
@@ -392,13 +402,15 @@ private String generateCode(AuthorizationRequest authorizationRequest, Authentic
         }
     }
 
+
+
     private String getSuccessfulRedirect(AuthorizationRequest authorizationRequest, String authorizationCode) {
 
         if (authorizationCode == null) {
             throw new IllegalStateException("No authorization code found in the current request scope.");
         }
 
-        UriComponentsBuilder template = UriComponentsBuilder.fromUriString(authorizationRequest.getRedirectUri());
+        UriComponentsBuilder template = uaaUrlUtils.parseAndDecodeUrl(authorizationRequest.getRedirectUri());
```

Comment:
We assume that the URI is encoded, so we properly decode it
this allows us to create a UriComponentsBuilder that can handle variable expansion